src/Features/CSharp/Portable/CodeRefactorings/UseRecursivePatterns/UseRecursivePatternsCodeRefactoringProvider.cs

```diff
@@ -295,7 +280,7 @@ static SyntaxToken Flip(SyntaxToken token)
                 .Where(d => d.Identifier.ValueText == identifierName.Identifier.ValueText)
                 .FirstOrDefault();
 
-            if (designation is not { Parent: PatternSyntax containingPattern })
+            if (designation is not { Parent: PatternSyntax(not SyntaxKind.ListPattern) containingPattern })
```

Comment:
=-o

i did not realize you could do that.

--

also, can you doc why this clause was added?

Reply:
Done. 

(this is using Deconstruct(SyntaxNode, out SyntaxKind))

Reply:
yup.  i just didn't realize that you could do the `not ...` bit in the position sectino :)  just never occurred to me :)